### DISCUSSION TITLE
[Berry] Find and replace string

### DISCUSSION THREAD

**Discussion by Slyke** [arendst/Tasmota#15805] [Q&A] [answered]
I noticed that the [Berry doc](https://github.com/Skiars/berry_doc/releases/download/latest/berry_short_manual.pdf) didn't have any functions for find and replace for strings.

So I made one and thought I'd share here:
```
def strReplace(haystack, needle, food)
  import string
  var output = ""
  var strSplit = string.split(haystack, needle)
  output = strSplit[0]
  for i: 1 .. (strSplit.size() - 1)
    output += food + strSplit[i]
  end

  return output
end
```

Usage and examples:
```
strReplace("hello", "ll", "xx")
> hexxo
strReplace("hellollo", "ll", "xx")
> hexxoxxo
strReplace("hellollo", "aa", "xx")
> hellollo
strReplace("hello", "ll", "")
> heo
strReplace("hello", "", "xx")
> hello
```

**Comment by sfromis** [comment, marked as answer]
For good measure, it would make sense to have the `import string` inside such a function, easier to move around than depending on a global import, like when having it part of a class.

This works too: `output += food + strSplit[i]`

**Reply by Slyke** [reply]
Done!

**Comment by s-hadinger** [comment]
Thanks for your contribution. I believe there is a simpler way that generates less creation of intermediary string objects. If that's proven equivalent, I will add it to the standard library:

``` berry
def strReplace(haystack, needle, food)
  import string
  return string.split(haystack, needle).concat(food)
end
```

**Comment by s-hadinger** [comment]
This is now part of the `string` library: `string.replace()`

See https://github.com/arendst/Tasmota/pull/15862
